software/contrib/preset_manager.py

```diff
@@ -0,0 +1,222 @@
+        self.selected_letter = 0
+        self.screen = 1
+
+        self.old_k2 = 0
+
+        self.resolution = 1000
+
+        self.box_width = int(OLED_WIDTH / 3)
+        self.box_height = int(OLED_HEIGHT / 2.8)
+
+        self.list_line_height = CHAR_HEIGHT + 4
+
+        self.output_multiplier = MAX_OUTPUT_VOLTAGE / self.resolution
+
+        b1.handler(self.back_button)
+        b2.handler(self.enter_button)
+
+        self.confirm_delete = False
+
+        self.characters = "aabcdefghijklmnopqrstuvwxyz0123456789  "
+        self.number_of_characters = len(self.characters)
+        self.max_characters = 16
+
+    def create_blank_preset(self, name):
+        return [str(name), [0, 0, 0, 0, 0, 0]]
+
+    def save_state(self):
+        """Save the current state variables as JSON."""
+        # Don't save if it has been less than 5 seconds since last save.
+        if self.last_saved() < 5000:
+            return
+        state = {"presets": self.presets, "loaded preset": self.loaded_preset}
+        self.save_state_json(state)
+
+    def back_button(self):
+        self.save_state()
+
+        self.screen -= 1
+
+        if self.screen == -3:  # Confirm delete
+            self.confirm_delete = True
+
+    def enter_button(self):
+        if self.screen < 1:
+            self.screen += 1
+        else:
+            None  # Furthest forward screen already reached
+
+        if self.screen == 1:
+            self.load_new_preset()
+
+        self.save_state()
+
+    def update_current_value(self, values):
+        old_selected_preset = self.selected_preset
+        self.selected_preset = k1.read_position(7) - 1
+        if self.selected_preset != -1:
+            if (
+                self.selected_preset != old_selected_preset
+            ):  # When selected preset is changed, take a snapshot of k2
+                self.old_k2 = k2.read_position(self.resolution)
+            else:
+                if (
+                    abs(k2.read_position(self.resolution, 512) - self.old_k2) > 4
+                ):  # k2 must change to allow value to be updated to prevent updating every value while scrolling
+                    values[self.selected_preset] = k2.read_position(self.resolution)
+        else:
+            None
+
+    def draw_current_preset(self):
+        oled.fill(0)
+
+        name = self.presets[self.loaded_preset][0]
+        values = self.presets[self.loaded_preset][1]
+
+        oled.text(name, int((OLED_WIDTH - (len(name) * CHAR_WIDTH)) / 2), 0, 1)
+
+        self.update_current_value(values)
+
+        for index, value in enumerate(values):
+            x = int((index % 3) * self.box_width)
+            y = int((index // 3) * self.box_height) + self.box_height - 1
+            oled.rect(
+                x, y, self.box_width, self.box_height, 1
+            ) if index != self.selected_preset else oled.fill_rect(
+                x, y, self.box_width, self.box_height, 1
+            )
```

Comment:
[required]: I would suggest using a normal if/else block instead of an inline. Reason being, the left side of the inline if/else will always get executed even if your `index != self.selected_preset` condition evaluates false. I also think the code will be easier to read that way.